lib/uniffi-zcash-cli/src/diff/git.rs

```diff
@@ -0,0 +1,50 @@
+use anyhow::Context;
+use git2::Repository;
+
+const LIBS_REPO_URL: &str = "https://github.com/zcash/librustzcash.git";
+
+// The way the git tag is constructed, assumes that the git tags
+// will always be in the format "lib-semver"
+//
+// NOTE: The tag is constructed in a format used by `librustzcash`, which is "libName-livVersion".
+// For example: zcash_primitives-0.10.2
+// So if the naming convention is not applied, the code will not be able to checkout the tag.
+pub(crate) fn init_libs_repo(
+    lib_name: &String,
+    repo_path: String,
+    version: String,
+) -> anyhow::Result<()> {
+    let tag = format!("{}-{}", lib_name, version);
+    let repo = match Repository::open(repo_path.to_owned()) {
+        Ok(r) => r,
+        Err(_) => {
+            Repository::clone(LIBS_REPO_URL, repo_path).context("failed to clone current repo")?
+        }
+    };
+
+    if repo.find_reference(&tag).is_err() {
+        checkout_tag(&repo, &tag)?;
+    }
+
+    Ok(())
+}
+
+// Chckout a certain tag from a given repository
+fn checkout_tag(repo: &git2::Repository, tag: &str) -> anyhow::Result<()> {
+    let (object, reference) = repo
+        .revparse_ext(tag)
+        .context(format!("git object \"{}\" not found", tag))?;
```

Comment:
so the issue with that line is that the `format!` macro will run even on `Ok` value, because you pass it by value to the context. The correct way to approach it would be to use the `with_context` and pass a function that will format an error, only if it happens. You will see this pattern also in other methods like `ok_or` vs `ok_or_else`
applies to other places as well